freqtrade/optimize/__init__.py

```diff
@@ -124,7 +125,7 @@ def make_testdata_path(datadir: str) -> str:
 
 def download_pairs(datadir, pairs: List[str],
                    ticker_interval: str,
-                   timerange: Optional[Tuple[Tuple, int, int]] = None) -> bool:
+                   timerange: TimeRange = TimeRange()) -> bool:
```

Comment:
Same as above

Reply:
honestly - no, I dropped it intentionally as all calls to these functions are made with timerange argument.
it's not needed as parse-timeframe always returns a `TimeRange` (or an exception) - and returns `starttype=None` and `stoptype=None` instead to show empty timeframes.

It's needed in a few places where timerange is not available/ called (i left it in those functions) - but other than that i think using Optional should be limited to where it's really needed.

I can readd it if you think it's an issue. 
but then i also need (unnecessary) "none"-checks within the function to make mypy happy